osu.Game/OsuGame.cs

```diff
@@ -315,8 +315,12 @@ private void load()
         /// The user should have already requested this interactively.
         /// </summary>
         /// <param name="beatmap">The beatmap to select.</param>
-        public void PresentBeatmap(BeatmapSetInfo beatmap)
+        /// <param name="findPredicate">Predicate used to find a difficulty to select</param>
+        public void PresentBeatmap(BeatmapSetInfo beatmap, Predicate<BeatmapInfo> findPredicate = null)
         {
+            // Use this predicate if non was provided. This will try to find some difficulty from current ruleset so we wouldn't have to change rulesets
```

Comment:
if this stays:
```suggestion
            // Use this predicate if none was provided. This will try to find some difficulty from current ruleset so we wouldn't have to change rulesets
```